snake-ladder-ui/src/Components/GameConfig/GameConfig.js

```diff
@@ -0,0 +1,149 @@
+  const [boardColumns, setBoardColumns] = useState(10);
+  const [diceCount, setDiceCount] = useState(1);
+  const [playerCount, setPlayerCount] = useState(2);
+  const [snake, setSnake] = useState([
+    { start: "", end: "" },
+  ]);
+  const [Ladder, setLadder] = useState([
+    { start: "", end: "" },
+  ]);
+
+  const handlePositionChangeSnake = (index, type, value) => {
+    const updatedSnakeOrLadder = [...snake];
+    updatedSnakeOrLadder[index][type] = value;
+    setSnake(updatedSnakeOrLadder);
+  };
+
+  const handleAddPositionSnake = () => {
+    setSnake([...snake, { start: "", end: "" }]);
+  };
+
+  const handlePositionChangeLadder = (index, type, value) => {
+    const updatedSnakeOrLadder = [...Ladder];
+    updatedSnakeOrLadder[index][type] = value;
+    setLadder(updatedSnakeOrLadder);
+  };
+
+  const handleAddPositionLadder = () => {
+    setLadder([...Ladder, { start: "", end: "" }]);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    console.log("Board Rows:", boardRows);
+    console.log("Board Columns:", boardColumns);
+    console.log("Dice Count:", diceCount);
+    console.log("Player Count:", playerCount);
+    console.log("Snake:", snake);
+    console.log("Ladder:", Ladder);
+  };
+
+  return (
+    <div className="container">
+      <h2 className="text-center mt-5">Game Settings</h2>
+      <form onSubmit={handleSubmit}>
+        <div className="mb-3">
+          <label htmlFor="boardRows" className="form-label">Board Rows:</label>
+          <input
+            type="number"
+            className="form-control"
+            id="boardRows"
+            value={boardRows}
+            onChange={(event) => setBoardRows(parseInt(event.target.value))}
+          />
+        </div>
+        <div className="mb-3">
+          <label htmlFor="boardColumns" className="form-label">Board Columns:</label>
+          <input
+            type="number"
+            className="form-control"
+            id="boardColumns"
+            value={boardColumns}
+            onChange={(event) => setBoardColumns(parseInt(event.target.value))}
+          />
+        </div>
+        <div className="mb-3">
+          <label htmlFor="diceCount" className="form-label">Dice Count:</label>
+          <input
+            type="number"
+            className="form-control"
+            id="diceCount"
+            value={diceCount}
+            onChange={(event) => setDiceCount(parseInt(event.target.value))}
+          />
+        </div>
+        <div className="mb-3">
+          <label htmlFor="playerCount" className="form-label">Player Count:</label>
+          <input
+            type="number"
+            className="form-control"
+            id="playerCount"
+            value={playerCount}
+            onChange={(event) => setPlayerCount(parseInt(event.target.value))}
+          />
+        </div>
+
+        <div className="mb-3">
+          <label className="form-label">Snake :</label>
```

Comment:
Here instead of creating separate fields for snake or ladder we can make only one field and based on values snake or ladder can we decided. We can have more discussion on this part.

Reply:
sure